api/v1alpha5/virtualmachine_hardware_types.go

```diff
@@ -17,6 +17,32 @@ const (
 	VirtualControllerTypeSATA VirtualControllerType = "SATA"
 )
 
+// MaxSlots returns the maximum number of slots per controller type.
+func (t VirtualControllerType) MaxSlots() int32 {
+	switch t {
+	case VirtualControllerTypeIDE:
+		return 2
+	case VirtualControllerTypeNVME,
+		VirtualControllerTypeSATA,
+		VirtualControllerTypeSCSI:
+		return 4
```

Comment:
Are you trying to define MaxSlotsPerBus? Otherwise the number seems duplicate against MaxPerVM